pages/docs/manual/latest/generate-converters-accessors.mdx

```diff
@@ -0,0 +1,565 @@
+---
+title: "Generate Converters & Helpers"
+description: "All about the @bs.deriving decorator, and how to generate code from types"
+canonical: "/docs/manual/latest/generate-converters-accessors"
+---
+
+# Generate Converters & Helpers
+
+<!-- TODO: genType -->
+
+When using ReScript, you will sometimes come into situations where you want to
+
+- Automatically generate functions that convert between ReScript's internal and JS runtime values (e.g. variants).
+- Convert a record type into an abstract type with generated creation, accessor and method functions.
+- Generate some other helper functions, such as functions from record attribute names.
+
+You can use the `@bs.deriving` decorator for different code generation scenarios. All different options and configurations will be discussed on this page.
+
+**Note:** Please be aware that extensive use of code generation might make it harder to understand your programs (since the code being generated is not visible in the source code, and you just need to know what kind of functions / values a decorator generates).
+
+## Generate Functions & Plain Values for Variants
+
+Use `@bs.deriving(accessors)` on a variant type to create accessor functions for its constructors.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res
+@bs.deriving(accessors)
+type action =
+  | Click
+  | Submit(string)
+  | Cancel;
+```
+
+```js
+function submit(param_0) {
+  return /* Submit */[param_0];
+}
+
+var click = /* Click */0;
+
+var cancel = /* Cancel */1;
+
+exports.click  = click;
+exports.submit = submit;
+exports.cancel = cancel;
+```
+
+</CodeTab>
+
+Variants constructors with payloads generate functions, payload-less constructors generate plain integers (the internal representation of variants).
+
+**Note**:
+- The generated accessors are lower-cased.
+- You can now use these helpers on the JavaScript side! But don't rely on their actual values please.
+- Doesn't work with polymorphic variants.
+
```

Comment:
it does not work with poly var, the reason is that polyvar does not need it, the encoding of poly var is standardised

Reply:
should i completely drop that line?

Reply:
dropped it